src/main/java/it/gov/pagopa/cgn/portal/service/AgreementService.java

```diff
@@ -0,0 +1,71 @@
+package it.gov.pagopa.cgn.portal.service;
+
+import it.gov.pagopa.cgn.portal.enums.AgreementStateEnum;
+import it.gov.pagopa.cgn.portal.exception.InvalidRequestException;
+import it.gov.pagopa.cgn.portal.model.AgreementEntity;
+import it.gov.pagopa.cgn.portal.model.AgreementUserEntity;
+import it.gov.pagopa.cgn.portal.repository.AgreementRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+@Service
+public class AgreementService {
+
+    private final AgreementRepository agreementRepository;
+
+    private final AgreementUserService userService;
+
+    public AgreementEntity findById(String subscriptionId) {
```

Comment:
```suggestion
    public AgreementEntity findById(String agreementId) {
```
Should we rename subscriptionId into agreementId while `findById` references to `Agreement` entity?

Reply:
Yes, I'll change it. Thanks